src/main/java/net/minecraftforge/fmp/capabilities/MultipartCapabilityHelper.java

```diff
@@ -0,0 +1,175 @@
+package net.minecraftforge.fmp.capabilities;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.IMultipartContainer;
+import net.minecraftforge.fmp.multipart.ISlottedPart;
+import net.minecraftforge.fmp.multipart.PartSlot;
+
+/**
+ * A general use multipart capability helper.
+ */
+public class MultipartCapabilityHelper
```

Comment:
What asie said. This is here to expand upon the (relatively limited) default cap system, as blocks may want to access the caps on only one edge of the block and not the whole TE. It's also much nicer than testing each one manually.